rapid7/meterpreter
# support emitting short filenames when listing files, fix windows stat path builder

I don't know how I missed it, but the path builder arguments are backward and use the wrong string type when we stat files. This fixes that, and adds a new TLV type so we can send down the short filenames when listing directories as well.

# Verification steps

Verification, similar to #137, but wait, it gets better! rapid7/metasploit-framework#4950 adds support for displaying the short filenames, so you can verify that they come through on Windows, and don't explode on Linux.

* [x] Passes post/test/meterpreter and post/test/file post modules on Windows 32-bit
* [x] Passes post/test/meterpreter and post/test/file post modules on Windows 64-bit
* [x] Passes post/test/meterpreter and post/test/file post modules on Linux
* [x] ls shows valid file sizes and types
* [ ] Short pathnames show up with the 'ls -x' option
```
meterpreter > cd c:
meterpreter > ls -x

Listing: C:\
============

Mode              Size      Type  Last modified              Short Name  Name
----              ----      ----  -------------              ----------  ----
40777/rwxrwxrwx   0         dir   2015-02-19 10:46:41 -0600              $Recycle.Bin
40777/rwxrwxrwx   0         dir   2015-02-19 09:31:32 -0600  $SYSRE~1    $SysReset
100666/rw-rw-rw-  1         fil   2013-06-18 07:18:29 -0500              BOOTNXT
40777/rwxrwxrwx   0         dir   2015-03-18 14:17:10 -0500              Config.Msi
40777/rwxrwxrwx   0         dir   2013-08-22 09:45:52 -0500  DOCUME~1    Documents and Settings
40777/rwxrwxrwx   0         dir   2013-08-22 10:22:35 -0500              PerfLogs
40555/r-xr-xr-x   0         dir   2015-03-17 10:29:10 -0500  PROGRA~1    Program Files
40555/r-xr-xr-x   0         dir   2015-03-18 14:05:34 -0500  PROGRA~2    Program Files (x86)
40777/rwxrwxrwx   0         dir   2015-03-18 14:21:01 -0500  PROGRA~3    ProgramData
40777/rwxrwxrwx   0         dir   2015-01-27 15:34:37 -0600              Python34
100666/rw-rw-rw-  0         fil   2015-02-19 09:56:39 -0600              Recovery.txt
40777/rwxrwxrwx   0         dir   2015-03-18 15:01:37 -0500  SYSTEM~1    System Volume Information
40555/r-xr-xr-x   0         dir   2015-02-19 10:00:34 -0600              Users
40777/rwxrwxrwx   0         dir   2015-03-18 14:31:01 -0500              Windows
40777/rwxrwxrwx   0         dir   2015-02-19 09:40:34 -0600              Windows.old
100444/r--r--r--  398356    fil   2014-03-18 05:14:54 -0500              bootmgr
40777/rwxrwxrwx   0         dir   2015-02-24 13:24:39 -0600              cygwin64
40777/rwxrwxrwx   0         dir   2015-02-18 23:18:39 -0600              found.000
40777/rwxrwxrwx   0         dir   2015-03-17 10:33:11 -0500  METASP~1    metasploit
0000/---------    0         fif   1969-12-31 18:00:00 -0600              pagefile.sys
100666/rw-rw-rw-  16777216  fil   2015-03-18 14:31:03 -0500              swapfile.sys
```
